posts/jhipster-deepdive-part1.md

```diff
@@ -0,0 +1,51 @@
+
+### Testing Frameworks
+By default, JHipster applications come bundled with two testing frameworks, JUnit and Karma.js, for Java unit tests and Javascript unit tests, respectively. Three other options are available to add for testing other aspects of your application; Gatling for performance tests, Cucumber for behavior-driven tests, and Protractor for Angular/React integration tests which simulate using the application like a real user.
+
+Once the application is generated, test files are placed in the `src/test` directory and are ran by executing `./mvnw clean test` (or `./gradlew test` for Gradle projects).
+
+### Modules
+
+Modules are other generators and plugins made by the community that offer integrations with third-party services or provide some other functionality to JHipster applications. A few of the popular modules include adding the [Ionic framework](https://ionicframework.com/), [PrimeNG user interface](https://www.primefaces.org/primeng/#/), and [React](https://reactjs.org/), which will be officially supported in the main JHipster generator in version 5.
+
+For a complete list of modules visit [the marketplace](https://www.jhipster.tech/modules/marketplace/#/list). You can also create and contribute your own module! Find the instructions for [creating a module here](https://www.jhipster.tech/modules/creating-a-module/) and check out [this article by Théo Lebrun](https://blog.ippon.tech/how-to-efficiently-use-jhipster-in-your-company-2/) where he explains how to create a module to customize JHipster to fit your needs.
+
+## Finishing Up
+Once you've finished picking your options, either in the command line interface or Web form, the scaffolded application will be generated. If the online Web form was used, simply extract the zip file to your chosen directory or pull the code from the linked GitHub repository. Run `yarn install` (or `npm install` if npm was chosen) to install dependencies; once completed, execute `./mvnw` for Maven projects or `./gradlew` for Gradle projects to build and run the backend, and in another terminal window in the same directory execute `yarn start` (or `npm start`) to run the frontend.
+
+> [Learn how to configure your IDE for JHipster here!](https://www.jhipster.tech/configuring-ide/)
+
+In the next post we will dive deep into JHipster's default tools and generating several entities with relationships of various complexities.
+
+---
+
+*At the time of this writing, the current version of JHipster is v4.14.4 and the next major version, v5.0, is in beta. Version 5 will bring official support for another frontend framework [React](https://reactjs.org/) among many improvements. A follow-up post will be written for JHipster 5's release. This series is written using a MacBook Pro running High Sierra. Some steps will be different for other operating systems.*
```

Comment:
JHipster 5 is coming soon, I think the article should target it directly.

Reply:
I hate to wait. That just gives us an excuse to write another blog. 😄